File: clojure/etl/etl.clj
(ns etl
  (:use [clojure.string :only [lower-case]]))

(defn transform
  "Transform a map of {key [value]} to {(lower-case value) key}"
  [input]
  (reduce-kv #(merge %1 (zipmap (map lower-case %3) (repeat %2))) {} input))

File: elixir/etl/etl.exs
defmodule ETL do
  def transform(old) do
    Enum.reduce(old, HashDict.new(), fn ({value, words}, dict) ->
      Enum.reduce(words, dict, fn (word, dict) ->
        # This will intentionally crash when a word is repeated (invalid input)
        HashDict.put_new(dict, String.downcase(word), value)
      end)
    end)
  end
end